pkg/controller/direct/monitoring/maputils.go

```diff
@@ -142,8 +142,16 @@ func Duration_FromProto(mapCtx *MapContext, in *durationpb.Duration) *string {
 		return nil
 	}
 
-	d := in.AsDuration()
-	s := fmt.Sprintf("%vs", d.Seconds())
+	// We want to report the duration without truncation (do don't want to map via float64)
```

Comment:
The issue we're fixing is that if translation to float64 products a number that is too big (uses exponents) or too small (nanosecond precision), it wasn't round-tripping.

In theory this is in the production code, in practice it doesn't make a difference for reasonable values.